### PR TITLE
chore: Deprecate Trial.logs in favor of Trial.iter_logs

### PR DESCRIPTION
**Pull request by wes-turner**
This renaming makes `logs` more consistent with the emerging `get_*` / `list_*` / `iter_*` standard.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/common/experimental/trial.py [hunk 1]
```diff
@@ -84,7 +84,16 @@ def __init__(self, trial_id: int, session: api.Session):
         self.summary_metrics: Optional[Dict[str, Any]] = None
         self.state: Optional[TrialState] = None
 
-    def logs(
+    def logs(self, *args: Any, **kwargs: Any) -> Iterable[str]:
```
Comment (azhou-determined) [inline harness/determined/common/experimental/trial.py:87]: something feels a little weird about changing the signature of this method and just passing everything through. passing a bogus arg won't get caught by this method anymore. 

eh, i don't suppose it makes any functional difference at all. and this does save us like 10 lines of code. 🤷

**Review by azhou-determined — APPROVED** [review]
lgtm

File: harness/determined/common/experimental/trial.py [hunk 2]
```diff
@@ -176,6 +187,8 @@ def logs(
         ):
             yield log.message
 
+    logs.__signature__ = inspect.signature(iter_logs)
```
Comment (azhou-determined) [inline harness/determined/common/experimental/trial.py:190]: can this be inside of logs somehow? weird to me that it's just floating in the file.

Comment (wes-turner) [reply harness/determined/common/experimental/trial.py]: No. It needs to run before `logs` is called. Moved `logs` to be closer, hope that helps.

**Pull request merged**